src/init.cpp

```diff
@@ -228,7 +228,6 @@ void PrepareShutdown()
 #endif
     GenerateBitcoins(false, 0, Params(), *g_connman);
```

Comment:
The dereference operator is still being called though in the function call even if the function doesn't make use of it.

This can result in undefined behavior if the stored pointer is null : http://www.cplusplus.com/reference/memory/unique_ptr/operator*/

> The unique_ptr shall not be empty (i.e., its stored pointer shall not be a null pointer) in order to be dereferenciable. 

I suggest this be fixed in another PR.